tests/Integration/Database/GenericTransactionManagerTest.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Integration\Database;
+
+use Exception;
+use PDOException;
+use Tempest\Database\Migrations\Migration;
+use Tempest\Database\Transactions\TransactionManager;
+use Tests\Tempest\Integration\FrameworkIntegrationTestCase;
+
+/**
+ * @internal
+ * @small
+ */
+final class GenericTransactionManagerTest extends FrameworkIntegrationTestCase
+{
+    public function test_transaction_manager(): void
+    {
+        $manager = $this->container->get(TransactionManager::class);
+
+        $manager->begin();
+
+        $this->console
+            ->call('migrate:up');
+
+        $this->assertCount(3, Migration::all());
+
+        $manager->rollback();
+
+        $this->expectExceptionMessage("no such table");
```

Comment:
I would assert on the exception message, but rather the exception class. If there's no dedicated class for this exception yet then feel free to add it.